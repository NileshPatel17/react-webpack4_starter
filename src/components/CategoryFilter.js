import React from 'react';
import cxs from 'cxs';
import PropTypes from 'prop-types';

const CategoryFilter = ({ active, name, onToggle }) => (
  <label
    className={cxs({
      marginRight: '.5rem',
      marginTop: '.5rem',
      borderRadius: '2px',
      backgroundColor: active ? '#ffd43b' : '#868e96',
      fontWeight: '500',
      color: '#fff',
      fontSize: '1em',
      lineHeight: '2em',
      display: 'inline-block',
      padding: '0 1em',
      cursor: 'pointer',
      whiteSpace: 'nowrap',
      textDecoration: 'none',
      borderWidth: '1px',
      borderStyle: 'solid',
      userSelect: 'none',
      outline: 'none'
    })}
  >
    <span
      className={cxs({
        color: active ? '#212529' : '#fff'
      })}
    >
      {name + ' '}
    </span>
    <input
      type="checkbox"
      checked={active}
      className={cxs({
        marginLeft: '.25em',
        verticalAlign: 'middle',
        position: 'relative',
        top: '-2px'
      })}
      onChange={e => onToggle(name, e.target.checked)}
    />
  </label>
);
CategoryFilter.propTypes = {
  active: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired,
  onToggle: PropTypes.func.isRequired
};
export default CategoryFilter;
