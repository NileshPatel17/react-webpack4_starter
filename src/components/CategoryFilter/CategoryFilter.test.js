import React from 'react';
import { shallow } from 'enzyme';

import CategoryFilter from './CategoryFilter.jsx';

describe('<CategoryFilter/>', () => {
  let wrapper;
  let mockOnToggle;
  const name = 'children';
  const checked = false;
  beforeEach(() => {
    mockOnToggle = jest.fn();
    const props = { active: checked, name: name, onToggle: mockOnToggle };
    wrapper = shallow(<CategoryFilter {...props} />);
  });

  it('CategoryFilter component should render as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render label text and input checkbox value correctly', () => {
    expect(
      wrapper
        .find('span')
        .text()
        .trim()
    ).toBe(name);
    expect(wrapper.find('input').props().checked).toBe(checked);
  });

  it('should checkbox active, when toggle', () => {
    const input = wrapper.find('input');
    input.simulate('change', { target: { checked: true } });
    expect(mockOnToggle).toHaveBeenCalled();
    // expect(input.props().checked).toBe(true);
  });
});
