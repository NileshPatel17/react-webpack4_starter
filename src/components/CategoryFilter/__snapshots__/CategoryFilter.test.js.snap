// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryFilter/> CategoryFilter component should render as expected 1`] = `
<label
  className="x0 x1 x2 x3 x4 x5 x6 x7 x8 x9 xa xb xc xd xe xf xg"
>
  <span
    className="x5"
  >
    children 
  </span>
  <input
    checked={false}
    className="xh xi xj xk"
    onChange={[Function]}
    type="checkbox"
  />
</label>
`;
